bson/bsoncodec/registry.go

```diff
@@ -56,60 +56,25 @@ var ErrNotInterface = errors.New("The provided type is not an interface")
 
 // A RegistryBuilder is used to build a Registry. This type is not goroutine
 // safe.
+//
+// Deprecated: Use Registry instead.
 type RegistryBuilder struct {
-	typeEncoders      map[reflect.Type]ValueEncoder
-	interfaceEncoders []interfaceValueEncoder
-	kindEncoders      map[reflect.Kind]ValueEncoder
-
-	typeDecoders      map[reflect.Type]ValueDecoder
-	interfaceDecoders []interfaceValueDecoder
-	kindDecoders      map[reflect.Kind]ValueDecoder
-
-	typeMap map[bsontype.Type]reflect.Type
-}
-
-// A Registry is used to store and retrieve codecs for types and interfaces. This type is the main
-// typed passed around and Encoders and Decoders are constructed from it.
-type Registry struct {
-	typeEncoders map[reflect.Type]ValueEncoder
-	typeDecoders map[reflect.Type]ValueDecoder
-
-	interfaceEncoders []interfaceValueEncoder
-	interfaceDecoders []interfaceValueDecoder
-
-	kindEncoders map[reflect.Kind]ValueEncoder
-	kindDecoders map[reflect.Kind]ValueDecoder
-
-	typeMap map[bsontype.Type]reflect.Type
-
-	mu sync.RWMutex
+	registry *Registry
```

Comment:
No, the goal is to push users to migrate from `RegistryBuilder` to `Registry` because `RegistryBuilder` will be removed in Go Driver 2.0. Addressing each point:
1. True, although not a significant benefit.
2. All of the APIs for actually using a custom registry (e.g. `MarshalWithRegistry`) only accept the `Registry` type (a concrete type, not an interface), so the additional encoder/decoder lookup functions provided by `Registry` would not be useful on the `RegistryBuilder` type.
3. We want users to completely migrate to `Registry` because we will remove the `RegistryBuilder` type in Go Driver 2.0. For now, we must retain the existing behavior of `RegistryBuilder` for backward compatibility only.